### PR TITLE
Various web UI fixes and improvements

### LINKED ISSUE #2402: ELRS AP drops out during ELRS AP scanning resulting in failure to save settings.
## Current Behavior

When the ELRS RX main configuration UI scans for access points the ELRS RX access point drops out.

This is very repeatable, as the screenshot below shows:

image

This doesn't happen if the unified target RX is reset to defaults by holding the button down for 8 seconds or so, as can be seen here:

image

## Steps to Reproduce

1. Flash unified target

2. Configure unified target

3. Connect to the ELRS RX AP but do NOT click 'Reconnect automatically'.

4. Start a ping test in a command prompt, e.g. `ping 10.0.0.1 -t`.

5. Visit http://10.0.0.1 in the browser.

6. Observe a few moments later the pings timeout and the OS will disconnect from the ELRS AP.

## Possible Solution (Not obligatory)

1. Make ELRS not drop the AP.
2. Show connection status in the UI.
3. Don't allow save if the connection status has dropped.

## Details

Although this testing is on a new PCB, I've observed this AP disconnect on my other ELRS kit (from memory: RM Boxer TX, HM EP RX's) but never dug into it till now.

When the AP drops out the user will not know, as the UI doesn't show a 'connected' state, so when the user tries to save the configuration when the AP has dropped out it will fail to save the configuration, as per this screenshot:

image

## Your Environment

- TX hardware: N/A
- RX hardware: SPRacing EVO-ELRSOSDVTX Rev B
- Handset model: N/A
- OpenTX version (including nightly number): N/A
- ExpressLRS version (TX & RX MUST MATCH): (0ef5b9 - https://github.com/spracing/ExpressLRS/tree/0ef5b9cc30062927e86317d894a7462b4cd28d89)
- Packet Rate: N/A
- Telemetry Ratio: N/A
- user_defines: N/A

### PR DESCRIPTION
**Pull request by pkendall64**
# 1. Improve display of `WiFi "auto on" interval` in web UI
## Problem
When flashing with `AUTO_WIFI_ON_INTERVAL` off in the configurator then going to the web-UI it displays a zero; if you then make changes and save, it will set the wifi-on-interval to that zero and will auto boot into wifi mode!

## Solution
When the interval is off in configurator, the interval is left out of the JSON, which is correct. This was not respected in the `options.cpp` code where it maintains an in-memory copy, and a divide caused a zero to be in the field.

# 2. Nicer UX around the bind-phrase
When entering a bind-phrase we now change the badge above the UID to show that it has been `Modified` and not been saved.
After saving, the bind-phrase is cleared and UID badge is updated to `Flashed` as per usual.

# 3. Add missing serial protocols
The DJI, SUMD and HoTT telemetry protocols were not handled and some were not even in the list!
These have been added and a few other UX cleanups have been done.
1. When airport is selected, you don't get the choice of protocol as airport is the mode of operation.
2. The `SBUS failsafe` section is only shown when choosing SBUS modes. i.e. SBUS, SBUS inverted, or DJI RS Pro 

# 4. Styled the Upload buttons!
After discussions with my web designer team-mate, she showed my how to style the upload buttons so she doesn't make puking sounds when she sees the UI... well at least not for that anyway 😄 

# 5. Deferred WiFi scanning
Only scan wifi networks when entering the wifi tab.
Fixes #2402

### CODE REVIEW

**Review by JyeSmith — APPROVED** [review]
Looking good.  Tested both tx and rx  👍

File: src/html/scan.js [hunk 1]
```diff
@@ -173,73 +177,38 @@ function init() {
   initOptions();
 }
 
-function changeCurrentColors() {
```
Comment (schugabe) [inline src/html/scan.js:176]: is this unused code? I can't find any reference to it

**Review by schugabe — APPROVED** [review]
- Tested the wifi change with disabled wifi
- UID badge works
- uploads button now have color
- serial protocol options are now available

The changes to the JS code are a bit hard to review but everything seems to be working.

We did already fix the wifi scanning issue once (since we did pile up too many requests). Did this issue appear again?

**Comment by pkendall64** [comment]
> We did already fix the wifi scanning issue once (since we did pile up too many requests). Did this issue appear again?

Yes we fixed it earlier, but this is not for a slightly different issue. i.e. when first loading the page it would do a wifi scan which would disconnect it as a client and if the user made a change and tried to save then it would fail. So this change does not start a wifi scan until they click on the wifi tab.

**Comment by hydra** [comment]
I'm sure this will lead to a much better end-user overall experience, LGTM!

**Pull request merged**